### PR TITLE
Port immutable.offloaded to Python 3

### PR DESCRIPTION
**Pull request by chadwhitacre**
https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3466

### Todo

- [x] set_status
- [x] timings
- [x] count
- [x] call

### CODE REVIEW

**Comment by itamarst** [comment]
I am not sure what the right answer here is, when 1/3rd of the file has no test coverage on Python 3. @exarkun @meejah any sense which tests give additional coverage to offloaded.py?

**Comment by itamarst** [comment]
Plan is for @exarkun to try to write more direct tests.

**Comment by chadwhitacre** [comment]
@exarkun @itamarst Should I close this PR? Leave it open for now? Merge?

If offloaded is undertested, that seems an issue under both 2 and 3. I'd lean towards merging this and addressing any bugs uncovered by additional testing when they are discovered.

**Comment by itamarst** [comment]
The main worry is "will this break on Python 2."

Insofar as test_system.py adds additional coverage, and that runs on Python 2... it's probably mergeable, pending review by @exarkun or @meejah.

**Review by itamarst — APPROVED** [review]

**Comment by exarkun** [comment]
I have some new tests written.  There are currently some problems with them on Python 3.6 though (see CI links from https://github.com/tahoe-lafs/tahoe-lafs/pull/868).  I'll follow-up with this on Monday.

**Comment by exarkun** [comment]
So it seems like the test failures associated with #868 are real Python 3 compatibility problems.  It looks like they're with parts of the module that this PR doesn't touch so they probably don't block this PR.  Instead, they show that there's more work to do porting this module.

I am fine with any of these courses:

1. Merge this.  Then, someone on the porting team takes over #868, merges master, and makes the tests pass on Python 3.
2. Combine this and #868 and make the tests pass on Python 3.
3. Remove test_helper from the ported list, finish #868 (which may not need anything more than that), merge #868, then merge master into this, re-mark test_helper as ported, and make the tests pass on Python 3.

File: src/allmydata/immutable/offloaded.py
Comment (exarkun) [inline src/allmydata/immutable/offloaded.py:144]: I don't know what this `str` was here for since `__full_version__` is already a str (ie, bytes) on Python 2 as far as I can tell.

The `.encode("utf-8")` seems fine and maybe it's required on Python 3 to make this bytes?  Since it is already a Python 2 str this is really an implicit ASCII decode followed by a UTF-8 encode which _could_ fail if the version has non-ASCII in it for some reason but it doesn't seem like it ever should.  I don't know if there's a better way to express "this string should be bytes on Python 2 and 3" and what's here.  If not, okay.  The extra decode will at least go away once Python 2 support is dropped.

Comment (chadwhitacre) [reply src/allmydata/immutable/offloaded.py:144]: I based this off of prior art:

https://github.com/tahoe-lafs/tahoe-lafs/blob/574c63d3505f66fd49497d3bf05490d45fc0aa57/src/allmydata/immutable/offloaded.py#L506-L509

**Comment by chadwhitacre** [comment]
Going for option (1) ... ☺️

**Pull request merged**
